src/components/Reports/Subscribers/SubscribersRedirect.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { useParams, Redirect } from 'react-router-dom';
+import SafeRedirect from '../../SafeRedirect';
+import { replaceSpaceWithSigns, extractParameter } from '../../../utils';
+import queryString from 'query-string';
+
+const SubscribersRedirect = ({ location }) => {
+  const { oldParam } = useParams();
+  const email = replaceSpaceWithSigns(extractParameter(location, queryString.parse, 'email'));
+  const section =
+    oldParam === 'subscriber-history' ? 'history' : oldParam === 'subscriber-gdpr' ? 'gdpr' : null;
+
+  if (!section) {
+    return <SafeRedirect to="/Lists/MasterSubscriber/" />;
+  }
+
+  return <Redirect to={`/subscribers/${email}/${section}`} />;
```

Comment:
Is it enough complete? I mean, are we sure that we do not need to keep the rest of the query string (page, language, etc)?

Reply:
If I'm not wrong, the only page that calls this pages only use email parameter.

Reply:
👍 

> If I'm not wrong

Please, be enough sure before continue.